rolldice.c

```diff
@@ -61,20 +61,27 @@ int rolldie ( int num_sides ) {
 /* parse_string() - Parses a string for dice rolling attributes
  *
  * Parameters: char *dice_string - string to parse
- * Returns: int * - array of nums describing the different aspects of the 
- * dice to be rolled
+ *             int *dice_nums - array of size DICE_ARRAY_SIZE
+ *                       where the parsed values will be stored
+ * Returns: int - return non-zero value if error occured
  */
-void parse_string(char *dice_string, int *dice_nums) {
+int parse_string(char *dice_string, int *dice_nums) {
     int temp_int = -1, res_int;
     const int DEFAULT_NUM_DICE = 1;
 
-    dice_nums[NUM_ROLLS] = 1;
-    dice_nums[NUM_DICE] = DEFAULT_NUM_DICE;
-    dice_nums[NUM_SIDES] = 6;
-    dice_nums[MULTIPLIER] = 1;
-    dice_nums[MODIFIER] = 0;
-    dice_nums[NUM_DROP] = 0;
-
+    if (*dice_string == '\0' && dice_nums[NUM_INITIALIZED]) {
```

Comment:
Since we're already using GNU Readline, might be nice to add in support for history for interactive mode instead of just "blank line == repeat".  I'm fine with getting this stuff in as an intermediate step, though, but if you can file an issue (once it's in) for adding full blown history support, I'd appreciate it.